workshops/pong_game/README.md

```diff
@@ -0,0 +1,213 @@
+    goto(x, y)
+    down()
+    begin_fill()
+    for count in range(2):
+        forward(width)
+        left(90)
+        forward(height)
+        left(90)
+    end_fill()
+```
+
+What to understand from this code:
+* We declare **rectangle** function with **x**, **y**, **width**, **height** as parameters for the function.
+* **up()** method is used to pull the turtle's pen up from the screen. It does not leave any marks or traces of its movements.
+* The turtle will go to `x`, `y` coordinates using the `goto()` method.
+* **down()** method will put the turtle's pen down.
+* **begin_fill()** tells the turtle that all upcoming closed graphical objects need to be filled with black color.
+* We define a **for loop** to create a rectangle of given `height` and `width`.
+* At last **end_fill()** tells turtle to stop the filling.
+
+![understanding gif](https://cloud-bbd2e31l2.vercel.app/0understanding_gif.gif)
+
+### 5.Draw game and move Pong Ball
+
+```python
+def draw():
+    clear()
+    rectangle(-200, state[1], 10, 50)
+    rectangle(190, state[2], 10, 50)
+
+    ball.move(aim)
+    x = ball.x
+    y = ball.y
+
+    up()
+    goto(x, y)
+    dot(10)
+    update()
+
+    if y < -200 or y > 200:
+        aim.y = -aim.y
+
+    if x < -185:
+        low = state[1]
+        high = state[1] + 50
+
+        if low <= y <= high:
+            aim.x = -aim.x
+        else:
+            return
+
+    if x > 185:
+        low = state[2]
+        high = state[2] + 50
+
+        if low <= y <= high:
+            aim.x = -aim.x
+        else:
+            return
+
+    ontimer(draw, 50)
+```
+
+![confusion gif](https://cloud-ea7bxmsh5.vercel.app/0confusion_gif.gif)
+
+Take a moment and try to understand what the code above is trying to say. The best way to interpret code is by turning it into plain English. Here are some helpful tips:
+* First of all we called the `clear()` function to tell Turtle to leave the current coordinates and without leaving any traces move to coordinates which will be given now onwards.
+* We called the `rectangle` function which we created before to draw **Paddle**. The first one will draw for player **A** and the second will draw for player **B**.
+* We will store the direction of `ball` in vector **aim**, which contains the *x* and *y* coordinates of the ball in each frame. Every time we update a new frame, simply we are adding the aim vector to the ball vector to get a new position.
+* Handling Collisions of **Pong Ball** is a very important part of code. So let us understand what we did next:
+    * The first **if** statement check if the ball hits the top or bottom of the screen. If it satisfies the condition, so simply it will reverse the **y** direction of the ball.
+    * Second **if** statement block check if ball hits **left** of the screen. Now, we first have to check if have hit the **paddle**. If we hit the paddle so simply **reverse** the **x** direction of the ball. If we did not hit the ball so **Game Over**.
```

Comment:
```suggestion
    * Second **if** statement block check if ball hits **left** of the screen. Now, we first have to check if have hit the **paddle**. If we hit the paddle so simply **reverse** the **x** direction of the ball. If we do not hit the ball, then it will be a **Game Over**.
```